src/ui/server/controllers/insurance/business/turnover/index.ts

```diff
@@ -29,51 +32,116 @@ const {
+  TURNOVER_ALTERNATIVE_CURRENCY_CHANGE,
+  TURNOVER_ALTERNATIVE_CURRENCY_CHECK_AND_CHANGE,
+  TURNOVER_SAVE_AND_BACK,
+  CREDIT_CONTROL,
+  CHECK_YOUR_ANSWERS,
+} = EXPORTER_BUSINESS_ROUTES;
 
 const { TURNOVER: TURNOVER_FIELDS } = FIELDS;
 
-const pageVariables = (referenceNumber: number) => ({
-  FIELDS: {
-    FINANCIAL_YEAR_END_DATE: {
-      ID: FINANCIAL_YEAR_END_DATE,
-      ...TURNOVER_FIELDS[FINANCIAL_YEAR_END_DATE],
-    },
-    ESTIMATED_ANNUAL_TURNOVER: {
-      ID: ESTIMATED_ANNUAL_TURNOVER,
-      ...TURNOVER_FIELDS[ESTIMATED_ANNUAL_TURNOVER],
-    },
-    PERCENTAGE_TURNOVER: {
-      ID: PERCENTAGE_TURNOVER,
-      ...TURNOVER_FIELDS[PERCENTAGE_TURNOVER],
+/**
+ * pageVariables for turnover page
+ * When is changeRoute, then alternative currency url should be ALTERNATIVE_CURRENCY_CHANGE
+ * when is checkAndChangeRoute, then alternative currency url should be ALTERNATIVE_CURRENCY_CHECK_AND_CHANGE
+ * else should be ALTERNATIVE_CURRENCY
+ * @param {Number} referenceNumber: Application reference number
+ * @param {Array<Currency>} currencies: Array of currencies
+ * @param {String} currencyCode: Provided currency code
+ * @param {Boolean} changeRoute: req.originalUrl is a change route
+ * @param {Boolean} checkAndChangeRoute: req.originalUrl is a check-and-change route
+ * @returns {Object} pageVariables
+ */
+const pageVariables = (referenceNumber: number, currencies: Array<Currency>, currencyCode: string, changeRoute?: boolean, checkAndChangeRoute?: boolean) => {
+  const currency = getCurrencyByCode(currencies, currencyCode);
+
+  let alternativeCurrencyUrl = `${INSURANCE_ROOT}/${referenceNumber}${TURNOVER_ALTERNATIVE_CURRENCY}`;
+
+  // if changeRoute, then URL should be ALTERNATIVE_CURRENCY_CHANGE
+  if (changeRoute) {
+    alternativeCurrencyUrl = `${INSURANCE_ROOT}/${referenceNumber}${TURNOVER_ALTERNATIVE_CURRENCY_CHANGE}`;
+  }
+
+  // if checkAndChangeRoute, then URL should be ALTERNATIVE_CURRENCY_CHECK_AND_CHANGE
+  if (checkAndChangeRoute) {
+    alternativeCurrencyUrl = `${INSURANCE_ROOT}/${referenceNumber}${TURNOVER_ALTERNATIVE_CURRENCY_CHECK_AND_CHANGE}`;
+  }
+
+  return {
+    FIELDS: {
+      FINANCIAL_YEAR_END_DATE: {
+        ID: FINANCIAL_YEAR_END_DATE,
+        ...TURNOVER_FIELDS[FINANCIAL_YEAR_END_DATE],
+      },
+      ESTIMATED_ANNUAL_TURNOVER: {
+        ID: ESTIMATED_ANNUAL_TURNOVER,
+        ...TURNOVER_FIELDS[ESTIMATED_ANNUAL_TURNOVER],
+      },
+      PERCENTAGE_TURNOVER: {
+        ID: PERCENTAGE_TURNOVER,
+        ...TURNOVER_FIELDS[PERCENTAGE_TURNOVER],
+      },
     },
-  },
-  PROVIDE_ALTERNATIVE_CURRENCY_URL: `${INSURANCE_ROOT}/${referenceNumber}${TURNOVER_ALTERNATIVE_CURRENCY}`,
-  SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${referenceNumber}${TURNOVER_SAVE_AND_BACK}`,
-});
+    PROVIDE_ALTERNATIVE_CURRENCY_URL: alternativeCurrencyUrl,
+    SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${referenceNumber}${TURNOVER_SAVE_AND_BACK}`,
+    CURRENCY_PREFIX_SYMBOL: currency.symbol,
+  };
+};
```

Comment:
Not a blocker for this PR, wondering if for such `pageVariables` functions like this when they're large, should we move to a new file inside the controller's directory

Reply:
This is happening in a couple of places now.  Would be worth a discussion to see if we can find a better solution for this, maybe just the way change and check and change is handled

Reply:
Definitely can extract `pageVariables` into it's own file, if it's more than a few lines. Agree worth a think about better ways to handle check/change in such pages